AI/VCAI/Engine/PriorityEvaluator.cpp

```diff
@@ -35,131 +37,28 @@ extern boost::thread_specific_ptr<VCAI> ai;
-		value->setRange(0.0, 1.0);
-
-		//we may want to use secondary hero(es) rather than main hero
-
-		//do not cancel important goals
-		//addRule("if lockedMissionImportance is HIGH then Value is very LOW");
-		//addRule("if lockedMissionImportance is MEDIUM then Value is somewhat LOW");
-		//addRule("if lockedMissionImportance is LOW then Value is HIGH");
-
-		//pick nearby objects if it's easy, avoid long walks
-		/*addRule("if turnDistance is SMALL then Value is somewhat HIGH");
-		addRule("if turnDistance is MEDIUM then Value is MEDIUM");
-		addRule("if turnDistance is LONG then Value is LOW");*/
-
-		//some goals are more rewarding by definition f.e. capturing town is more important than collecting resource - experimental
-		addRule("if turnDistance is MEDIUM then Value is MEDIUM");
-		addRule("if turnDistance is SMALL then Value is BITHIGH");
-		addRule("if turnDistance is LONG then Value is BITLOW");
-		addRule("if turnDistance is very LONG then Value is LOW");
-		addRule("if goldReward is LOW then Value is MEDIUM");
-		addRule("if goldReward is MEDIUM and armyLoss is LOW then Value is BITHIGH");
-		addRule("if goldReward is HIGH and armyLoss is LOW then Value is HIGH");
-		addRule("if armyReward is LOW then Value is MEDIUM");
-		addRule("if armyReward is MEDIUM and armyLoss is LOW then Value is BITHIGH");
-		addRule("if armyReward is HIGH and armyLoss is LOW then Value is HIGHEST with 0.5");
-		addRule("if armyReward is HIGH and goldReward is HIGH and armyLoss is LOW then Value is HIGHEST");
-		addRule("if armyReward is HIGH and goldReward is MEDIUM and armyLoss is LOW then Value is HIGHEST with 0.8");
-		addRule("if armyReward is MEDIUM and goldReward is HIGH and armyLoss is LOW then Value is HIGHEST with 0.5");
-		addRule("if armyReward is MEDIUM and goldReward is MEDIUM and armyLoss is LOW then Value is HIGH");
-		addRule("if armyReward is HIGH and turnDistance is SMALL and armyLoss is LOW then Value is HIGHEST");
-		addRule("if goldReward is HIGH and turnDistance is SMALL and armyLoss is LOW then Value is HIGHEST");
-		addRule("if armyReward is MEDIUM and turnDistance is SMALL and armyLoss is LOW then Value is HIGH");
-		addRule("if goldReward is MEDIUM and turnDistance is SMALL and armyLoss is LOW then Value is BITHIGH");
-		addRule("if goldReward is LOW and armyReward is LOW and turnDistance is not SMALL then Value is LOWEST");
-		addRule("if armyLoss is HIGH then Value is LOWEST");
-		addRule("if armyLoss is LOW then Value is MEDIUM");
-		addRule("if armyReward is LOW and turnDistance is LONG then Value is LOWEST");
-		addRule("if danger is NONE and armyStrength is HIGH and armyReward is LOW then Value is LOW");
-		addRule("if danger is NONE and armyStrength is HIGH and armyReward is MEDIUM then Value is BITLOW");
-	}
-	catch(fl::Exception & fe)
-	{
-		logAi->error("visitTile: %s", fe.getWhat());
-	}
+	auto file = CResourceHandler::get("initial")->load(ResourceID("config/ai-priorities.txt"))->readAll();
```

Comment:
Is it intended that by using `initial` filesystem ai-priorities is forbidden to be altered by mods?

Reply:
I know nothing about modding. First of all I need this file to be copied correctly during build. It is not the case now. If you can guide me how to use vcmi filesystem I will be happy. I expect it has to be added to some cmake? Also I need new resource type *.fll (fuzzy light language)

Reply:
* config folder being copied completely, your file it should be handled too.
* you should add `fll` extension to `EResType` or use `*.txt` - you feed a string to importer anyway.
* for now you could just rename to txt and load with
```
CResourceHandler::get()->load(ResourceID("config/ai-priorities.txt", EResType::TEXT))->readAll();
```

Reply:
Fll is automatically opened by fuzzy light client. I can use txt so. Is the place for file correct? Can I put it to a nee folder "ai"?

Reply:
'config/ai' is a good place

Reply:
Ho to access it in resource manager? Same config/priorities.txt or config/ai/priorities.txt?

Reply:
config/ai/priorities.txt